### DISCUSSION TITLE
Is there a way to transform a resource attribute to a metric label?

### DISCUSSION THREAD

**Discussion by singku** [open-telemetry/opentelemetry-collector-contrib#12656] [General]
We had a requirement to generate a new metric out of resource attribute then remove the resource attribute from the original metrics. Is there a way we can do it?

**Comment by spurge** [comment]
You could in theory add an attribute using the [metricstransform](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/metricstransformprocessor) or the [attributes](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/attributesprocessor) processors from the resource and then remove the resource with the [resource](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourceprocessor) processor.
